### DISCUSSION TITLE
I can't understand the logic

### DISCUSSION THREAD

**Discussion by JohnnyMaynne** [spatie/laravel-stats#40] [Q&A]
Hello everyone 😊

Please help me because I just can’t understand why I am receiving this data.

Here is a list of my records sorted from oldest to most recent: 

<img width="658" alt="image" src="https://github.com/user-attachments/assets/535f6b03-5df6-4024-95e8-27908fb936cf">

As you can see, there are only **9 records** and all of them were created in August from 09 to 18.  

During this period, **2 records were deleted** and this can be seen in the statistics.

 So, requesting the following data:

<img width="501" alt="image" src="https://github.com/user-attachments/assets/9ac538ba-5a58-4f2b-a228-e068849f0a54">

I get this result:

<img width="509" alt="image" src="https://github.com/user-attachments/assets/d9e2b10b-df24-4bf2-886a-d089629e947b">

So here it is. Why do I get the **value 5**?  After all, at the end of the period I have **7 entries**, the same thing will happen if I try other types of grouping. 

Even if I choose to group by day, the value will still be 5.

**Comment by JohnnyMaynne** [comment]
No ideas?

**Comment by sjardim** [comment]
The logic IMHO is:

Although 9 events of type HeadCountStats were fired in the period above, the state of "head counts" is 5. You had 7 "head counts" for a while, but 2 got 'deleted' so the rest is 5. 

In Event Sourcing, the events change the state of something and this package is a simple implementation of this technique. 

If all events had been "HeadCountStats::increase()",  the "value" would be 9 for that period.

**Comment by JohnnyMaynne** [comment]
> Логика, ИМХО, такова:
> 
> Хотя за указанный период было запущено 9 событий типа HeadCountStats, состояние "количество сотрудников" равно 5. Какое-то время у вас было 7 "подсчетов голов", но 2 были "удалены", так что осталось 5.
> 
> В Event Sourcing события изменяют состояние чего-либо, и этот пакет является простой реализацией этого метода.
> 
> Если бы все события были "HeadCountStats::increase()", "значение" было бы равно 9 за этот период.

Based on this logic, the data that the packet generates is incorrect, because it should not just be the sum of all events for the period

**Comment by EriBloo** [comment]
I would like to join and help, but I don't see any problem here. Data seems correct. What would you expect the data to be given your inputs?
